packages/store/addon/-private/system/model/record-data.ts

```diff
@@ -57,7 +76,9 @@ export default class RecordDataDefault implements RelationshipRecordData {
   // PUBLIC API
 
   getResourceIdentifier(): RecordIdentifier {
-    return this.identifier;
+    return IDENTIFIERS
```

Comment:
Did we use to have `lid` in the non feature flag case?

Reply:
it get's impossibly hard to type the identifiers if we have a branch that doesn't have `lid` or which makes `lid` optional.